vcd/resource_vcd_vapp_nat_rules.go

```diff
@@ -152,13 +158,19 @@ func resourceVappNetworkNatRulesUpdate(d *schema.ResourceData, meta interface{})
 	if !d.Get("enable_ip_masquerade").(bool) && natType == portForwardingNatType {
 		policy = allowTrafficPolicy
 	}
-	vappNetwork, err := vapp.UpdateNetworkNatRules(networkId, netRules,
+	vappNetwork, err := vapp.UpdateNetworkNatRules(networkId, netRules, d.Get("enabled").(bool),
 		natType, policy)
 	if err != nil {
 		log.Printf("[INFO] Error setting NAT rules: %s", err)
 		return fmt.Errorf("error setting NAT rules: %s", err)
 	}
 
+	if vappNetwork.Configuration.Features.FirewallService != nil &&
+		!vappNetwork.Configuration.Features.FirewallService.IsEnabled &&
+		d.Get("enabled").(bool) {
+		_, _ = fmt.Fprint(getTerraformStdout(), "WARNING: to enable NAT, firewall has to be enabled. Please use vcd_vapp_firewall_rules.enabled to enable the firewall.\n")
```

Comment:
One small polish, as I just realized that FW is enabled by default. Sorry for crunchiness and thanks. Please adjust for read warning as well.

```suggestion
		_, _ = fmt.Fprint(getTerraformStdout(), "WARNING: to enable NAT, firewall has to be enabled. Please use vcd_vapp_firewall_rules to enable the firewall\n")
```

Reply:
Newer suggestion:
`WARNING: for NAT rules to work, firewall has to be enabled. Please use vcd_vapp_firewall_rules to enable it\n`

CC: @Didainius

Reply:
Ended up even less prescriptive. Thanks!
`WARNING: for NAT rules to work, firewall has to be enabled. It can be enabled using vcd_vapp_firewall_rules`